tenant.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package goosechase.data.tenant;
+
+option go_package = "./proto";
+
+message Tenant {
+  string id = 1;
+  string name = 2;
+  string description = 3;
+  string domain = 4;
+  string domain_verification_status = 5;
```

Comment:
I did some refactoring on this in db level
https://github.com/goosechaseapp/sql/blob/ccafca81aac03e0dc2134ab177034f828d5b805b/migrations/20240425153831_initial.sql#L11C1-L19C3

Basically now domain data is stored in a separate table, which imo make schema more clean + add support for multiple domains per tenant in future. If you are agree with this change, wouldn't it make more sense to push domain related data to a separate message?